include/caffe/net.hpp

```diff
@@ -52,6 +54,19 @@ class Net {
    * included.
    */
   Dtype ForwardFromTo(int start, int end);
+  /**
+   * Basically, a ForwardFromTo, that release Python GIL, so that pycaffe
+   * can make multithreaded predictions. This is exposed to python-boost interface
+   * instead of ForwardFromTo
+   */
+  Dtype PyForwardFromTo(int start, int end) {
+    // Release GIL
+    m_thread_state = PyEval_SaveThread();
```

Comment:
RAII wrapper: Thank you for the suggestion. I will take a look to fix it.

Not right place: I agree. However, where would you suggest to place it? (consider that you have to reacquire GIL in case you forward/backward a python layer within the net). Even though this solution insert "Pythonish" code within the C++ logic, not releasing GIL for the pycaffe is a painful bug for python interface. If there is a cleaner alternative I would love to discuss about it.